tac/aea/mail/protocol.py

```diff
@@ -175,41 +181,41 @@ def __eq__(self, other):
             and self.protocol_id == other.protocol_id \
             and self._message == other._message
 
-    def encode(self, encoder: Encoder) -> bytes:
+    def encode(self) -> bytes:
         """
         Encode the envelope.
 
-        :param encoder: the encoder to use in order to encode the body of the envelope.
         :return: the encoded envelope.
         """
         envelope = self
         envelope_pb = base_pb2.Envelope()
```

Comment:
Aren't these required args?; why do we not force them to be present and throw an error if not?